### PR TITLE
Integrating dev and column creation

### PR DESCRIPTION
**Pull request by swathivottigunta**
Integrating dev (authentication) and column creation(add/edit card, add column, fetching the board detail from db)

- [x] Add column details to the board without refreshing the page

- [ ] Change add card modal dialog to child component of column

### CODE REVIEW

File: server/routes/api/cards.js [hunk 1]
```diff
@@ -1,76 +1,118 @@
...
+    Card.findByIdAndUpdate(req.params.id, req.body, (err, updatedCard) => {
+      if (!updatedCard) return res.status(400).send({ msg: "Invalid Card" });
+
+      console.log("updatedCard", updatedCard);
+      res.send(updatedCard);
+    });
+  } catch (err) {
+    res.status(500).send("Server Error");
+  }
+});
```
Comment (nicktran6997) [inline server/routes/api/cards.js:59]: Curious about where you're planning on calling this route. Is it going to be invoked from the frontend?

File: client/src/components/forms/props.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        name: 'desc',
+        placeholder: 'Type your description here',
+      },
+    },
+    addCard: {
+      title: {
+        id: 'name',
+        label: 'Add Title *',
+        name: 'name',
+        type: 'text',
```
Comment (nicktran6997) [inline client/src/components/forms/props.js:33]: For clarification, when you create a **new** card, you can only add a title but not a description as well? Would it be better to be able to add a description as well?

Comment (nicktran6997) [reply client/src/components/forms/props.js:33]: If a description is being added during card creation, then nvm about the above; just gonna note that most editing placeholders use the original description as a placeholder

Comment (benihime1458) [reply client/src/components/forms/props.js:33]: You know... I actually find it weird that a card needs a "title".  But to be clear I assume "title" is just the text that shows on the card when looking at the board-ui but when you click on the card to edit it, then you can see a more detailed description in "desc". 

Here's a visual of what that's like in Trello. 
https://imgur.com/a/jMrhJwe

So it doesn't look like you add the description on card creation but only the title.

File: client/src/components/dashboard/AddColumn.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  const {boardId, setLoadBoard} = props;
+  const [open, setOpen] = useState(false);
+
+  const handleClickOpen = (event) => {
+    event.preventDefault();
+    setOpen(true);
+  };
+  return (
+    // eslint-disable-next-line react/jsx-fragments
+    <Fragment>
```
Comment (ugur-akin) [inline client/src/components/dashboard/AddColumn.js:17]: Please try not to disable eslint rules. They help enforce a cleaner style more often than not. In this particular case, [this eslint rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-fragments.md) wants you to use [React's shorthand syntax for `React.Fragment `components](https://reactjs.org/docs/fragments.html#short-syntax).

File: client/src/components/dashboard/Card.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  };
+
+  const handleClickOpen = () => {
+    getCardData();
+    if (!cardError) setOpen(true);
+  };
+
+  return (
+    // eslint-disable-next-line react/jsx-fragments
+    <Fragment>
```
Comment (ugur-akin) [inline client/src/components/dashboard/Card.js:27]: Here as well.

File: server/routes/api/cards.js (hunk 1 above)
Comment (swathivottigunta) [reply server/routes/api/cards.js:59]: This endpoint is meant for detail edit card dialog where we edit the desc and save

File: client/src/components/dashboard/AddColumn.js (hunk 3 above)
Comment (swathivottigunta) [reply client/src/components/dashboard/AddColumn.js:17]: I dont remember exactly what I went through. It was the last moment fix. When I was trying quick fix, Fragment component changed to fragments (not React.Fragment). And the fragments appeared on my browser console.  Thats why I disabled it. I will consider your point next time. Thanks for letting me know

**Comment by swathivottigunta** [comment]
Refer: Featuring calendar and multiple branch

**Pull request closed without merging**